### PR TITLE
interpreter exec refactor

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Here is a proposed refactor ...

1. the binop computer now uses the instruction names, rather than JavaScript 'signs', i.e. '+' becomes 'add', removing a further look-up
2. kernel/exec has been refactored so that all binary operations share the same 'case' block.

I committed by own benchmark result before the refactor. Here is the mean time before and after:

mean 474.9725275000 ns
mean 481.3720141000 ns

If everyone is happy with this, I'll look at a similar change for unary operations.

(before merging, I'll remove the benchmark commit, and tidy up a few comments)

relates to #56

### CODE REVIEW

**Comment by ColinEberhardt** [comment]
Travis has fallen over again, during `npm install`

**Comment by maurobringolf** [comment]
👍  For using the `instruction.id` all the way through, its more consistent like this.

File: src/interpreter/kernel/exec.js [hunk 1]
```diff
@@ -678,328 +678,58 @@ export function executeStackFrame(frame: StackFrame, depth: number = 0): any {
...
+    case 'sub':
+    case 'div_s':
+    case 'div_u':
+    case 'div':
+    case 'min':
+    case 'max':
+    case 'copysign':
+    case 'or':
+    case 'xor': {
+      let binopComputer;
```
Comment (xtuc) [inline src/interpreter/kernel/exec.js]: Sounds a bit confusing IMO. What about binopFn or binopCalculator?

File: benchmark/add/results [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-mean 21.7151342000 ns
+total 255.1195230000 ms
+mean 25.5119523000 ns
 
 Interations: 10000000
-Date: 2017-12-29
+Date: 2018-1-6
 V8 version: 6.2.414.46-node.15
-Interpreter version: 0.0.14
+Interpreter version: 0.0.15
```
Comment (xtuc) [inline benchmark/add/results]: We can totally merge this file, but I should probably run it from my machine to keep it consistent, your native result seems a bit slower.

**Comment by ColinEberhardt** [comment]
> Sounds a bit confusing IMO. What about binopFn or binopCalculator?

OK, I've gone for `binopFn`. I've also rebased out my benchmark results. Should be good to go now.

**Comment by xtuc** [comment]
Here are the results: https://github.com/xtuc/js-webassembly-interpreter/commit/32a5fd2158081a1416d505d25160e3f26ac782a9 :+1:

**Pull request merged**